ballerina/schemaClient.bal

```
@@ -0,0 +1,300 @@
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+import ballerina/sql;
+
+# Represents an SQL metadata client.
+isolated client class SchemaClient {
+    private final Client dbClient;
+    private final string database;
+
+    # Initializes the SchemaClient object
+    #
+    # + host - The name of where the mysql server is hosted (ex: localhost)
+    # + user - The username to access the database
+    # + password - The password to access the database
+    # + database - The name of the database to be accessed
+    # + return - A `sql:Error` or nil
+    public function init(string host, string user, string password, string database) returns sql:Error? {
+        self.database = database;
+        self.dbClient = check new (host, user, password);
+    }
+
+    # Retrieves all tables in the database.
+    #
+    # + return - A string array containing the names of the tables or an `sql:Error`
+    isolated remote function listTables() returns string[]|sql:Error {
+        string[] tables = [];
+        stream<record {|string table_name;|}, sql:Error?> results = self.dbClient->query(
+            `SELECT TABLE_NAME FROM INFORMATION_SCHEMA.TABLES 
+             WHERE TABLE_SCHEMA = ${self.database};`
+        );
+
+        do {
+            check from record {|string table_name;|} result in results
+                do {
+                    tables.push(result.table_name.toString());
+                };
+        } on fail error e {
+            return error(string `Error while listing the tables in  the ${self.database} database.`, cause = e);
+        }
+
+        do {
+            check results.close();
+        } on fail error e {
+            return error("Error while closing the result stream.", cause = e);
+        }
```

I feel like we can simply return the error itself instead of catching and wrapping it

```suggestion
        check results.close();
```

---

@daneshk @niveathika